### PR TITLE
Add support for Push Constants in PipelineLayout

### PR DESCRIPTION
**Pull request by nipunG314**
I want to use Push Constants in a sample I'm working on, and would like to add Push Constants to PipelineLayout for the same.

Are these changes sufficient for this?

### CODE REVIEW

File: vulkan_helpers/vulkan_application.h [hunk 1]
```diff
@@ -259,14 +260,19 @@ class PipelineLayout {
           CreateDescriptorSetLayout(allocator, device, binding_list.bindings_, binding_list.flags_));
       raw_layouts.push_back(descriptor_set_layouts_.back());
     }
+
+    containers::vector<VkPushConstantRange> raw_push_ranges(ranges.begin(),
```
Comment (AWoloszyn) [inline vulkan_helpers/vulkan_application.h:264]: Looks reasonable, I would prefer this to be called push_constant_ranges.

Comment (nipunG314) [reply vulkan_helpers/vulkan_application.h]: Done

File: vulkan_helpers/vulkan_application.h [hunk 2]
```diff
@@ -247,7 +247,8 @@ class PipelineLayout {
  private:
   // TODO(awoloszyn): Handle push constants here too
```
Comment (krmoule) [inline vulkan_helpers/vulkan_application.h:248]: I think this comment can be removed now.

**Comment by nipunG314** [comment]
@AWoloszyn Since I'll be using this change for the overlapping_frames sample, could you please rebase the commit history at this point into the google/overlapping_frames staging branch? Then, the commit history between this repo and my fork would be aligned.

**Comment by AWoloszyn** [comment]
done

**Comment by nipunG314** [comment]
I'm sorry. I just realized I missed a detail in the PR.

The class PipelineLayout has a private constructor. I need to add the push_constant_ranges to the VulkanApplication function CreatePipelineLayout in order to actually use it. I'll put a new PR for the same.

**Pull request merged**